src/lib/Microsoft.Health.Events/EventConsumers/Service/EventBatchingService.cs

```diff
@@ -123,7 +129,7 @@ private async Task ThresholdWaitReached(string partitionId, DateTime windowEnd)
             {
                 _logger.LogTrace($"Partition {partitionId} threshold wait reached. Flushing {_eventPartitions[partitionId].GetPartitionBatchCount()} events up to: {windowEnd}");
                 var events = await GetPartition(partitionId).Flush(windowEnd);
-                await CompleteProcessing(events);
+                await CompleteProcessing(partitionId, events, triggerReason: nameof(ThresholdWaitReached));
```

Comment:
I think it was mentioned at one point that if the partition does not have any events, then a MaximumWaitEvent would never be generated.

However when recently testing with an event hub with no events, I was able to see a MaximumWaitEvent generated after 60 seconds. One thing I did notice however was that CompleteProcessing in ThresholdWaitReached is not called due to

`if (windowEnd < DateTime.UtcNow.AddSeconds(_timeBuffer))`

which is expected for out current use case for maximum wait (don't want to flush events unless we are outside the time window, so we wait until we are outside the time window before flushing).

In any case if we want our metrics to indicate that a partition with no events to show that we are "up-to-date" when we receive a maximum wait event and there are no events to process for that partition then I think we can use MaximumWaitEvent - we just may need to modify ThresholdWaitReached to log your new data freshness metric (or re-arrange how CompleteProcessing works when the is a maximumwaitevent and no real events)

Reply:
Yep when I did some digging on that, I did notice MaximumWaitEvent was being emitted for all partitions (regardless of if there was any data event in these partitions). The reason the chart/graph was indicating that was because of that check, which I did fix by calling LogDataFreshness() in the else part. Seems like that is the same you are suggesting, but let me know if that is not enough in line 87 - 
`
else
          {
                    // If we received the timer event and there are no enqueued events in the partition, then simply update the data freshness. 
 LogDataFreshness(partitionId, triggerReason: nameof(ThresholdWaitReached));
`

Reply:
ah, thanks, I missed that, looks good.